openlabcmd/openlabcmd/plugins/jobs/image.py

```diff
@@ -15,6 +16,9 @@ def check(self):
         image_check = 'openstack --os-cloud %s image list ' \
                       '-c "Name" -c "Status" -f value | grep cirros' % self.cloud
         res = subprocess.getoutput(image_check)
-        if "cirros" not in res:
+        if "cirros-0.3.5" not in res:
             self.failed = True
-            self.reasons.append("- Image: cirros image not found.")
+            self.reasons.append(Recover.IMAGE_CIRROS_035)
+        if "cirros-0.4.0" not in res:
```

Comment:
First, if cirros-0.4.0 image in public cloud, we append the reason Recover.IMAGE_CIRROS_035, and if --recover specified, then we will download and register cirros 0.3.5 image for the cloud? Same case if there is only cirros-0.3.5 in cloud. Second, if there is no cirros image in cloud, we append two reasons, then if --recover specified, the cloud will download and register two cirros images, right? I don' t know whether it's ok, see the code like "_IMAGE=$(openstack image list | grep -i cirros | head -n 1)" not sure which one it will return and the test expected.

Reply:
I understand your worries, but I think if the commands with "head -n 1" means the job just want a image but don't need a specific version image, so it's not a problem.

But I only saw a problem line [1] in cirros search results[2]:

> src_img_name=$(openstack image list -f value -c Name | grep -Eo "cirros-.*" | grep -v ramdisk | grep -v kernel)

I think we need add a `head -n1` for it?

[1] https://github.com/theopenlab/openlab-zuul-jobs/blob/97b0fe4b/playbooks/packer-functional-devstack/run.yaml#L19
[2] https://github.com/theopenlab/openlab-zuul-jobs/search?p=1&q=cirros&unscoped_q=cirros

Reply:
And `openstack image show` has potential duplicate image problem(like `More than one resource exists with the name or ID 'cirros-0.3.5-x86_64-disk'.`)

So, maybe we should change all the image get script to "openstack image list | grep xxx | head -n 1"?

Reply:
Sorry, I am not sure, please check them. And not sure if we can use cirros-0.4.0 only?